rust-bot/src/main.rs

```diff
@@ -38,6 +38,22 @@ fn ping(ctx: &mut Context, msg: &Message) -> CommandResult {
     return Ok(());
 }
 
+#[instrument(skip(ctx))]
+#[command]
+fn pin(ctx: &mut Context, msg: &Message) -> CommandResult {
+    let message_id = msg.content.parse::<u64>()?;
+    let result = ctx
+        .http
+        .get_message(*msg.channel_id.as_u64(), message_id)
+        .and_then(|rmsg| rmsg.pin(&ctx.http))
+        .and_then(|_| msg.channel_id.say(&ctx.http, "Pinned! :shadescorgi:"));
+
+    if let Err(e) = result {
+        println!("{}", e);
+    }
+    Ok(())
```

Comment:
if we return an error here, we can handle it in the after function apparently (TIL)

Reply:
🤯 Oh that makes so much sense! Another lovely blog post.